codec_native.go

```diff
@@ -138,10 +138,12 @@ func createEncoderOfNative(schema Schema, typ reflect2.Type) ValEncoder {
 		return &int16Codec{}
 
 	case reflect.Int32:
-		if schema.Type() != Int {
-			break
+		if schema.Type() == Long {
+			return &int32LongCodec{}
+		} else if schema.Type() == Int {
```

Comment:
The else is a code smell. Rather make this a switch like:
```
switch schema.Type() {
case ...:
    ...
case ....:
   ....
}
```

Reply:
Ok makes sense. Fixed 👍